pulsar-discovery-service/src/main/java/com/yahoo/pulsar/discovery/service/BrokerDiscoveryProvider.java

```diff
@@ -33,26 +38,32 @@
 import com.yahoo.pulsar.common.policies.data.loadbalancer.LoadReport;
 import com.yahoo.pulsar.discovery.service.server.ServiceConfig;
 import com.yahoo.pulsar.discovery.service.web.ZookeeperCacheLoader;
+import com.yahoo.pulsar.zookeeper.GlobalZooKeeperCache;
 import com.yahoo.pulsar.zookeeper.ZooKeeperCache.Deserializer;
 import com.yahoo.pulsar.zookeeper.ZooKeeperClientFactory;
 
 /**
  * Maintains available active broker list and returns next active broker in round-robin for discovery service.
  *
  */
-public class BrokerDiscoveryProvider {
+public class BrokerDiscoveryProvider implements Closeable {
 
-    protected ZookeeperCacheLoader localZkCache;
-    protected ZookeeperCacheLoader globalZkCache;
+    final ZookeeperCacheLoader localZkCache;
+    final GlobalZooKeeperCache globalZkCache;
     private final AtomicInteger counter = new AtomicInteger();
-    
+
+    private final OrderedSafeExecutor orderedExecutor = new OrderedSafeExecutor(4, "pulsar-discovery-ordered");
+    private final ScheduledExecutorService scheduledExecutorScheduler = Executors.newScheduledThreadPool(1);
+
     private static final String PARTITIONED_TOPIC_PATH_ZNODE = "partitioned-topics";
 
     public BrokerDiscoveryProvider(ServiceConfig config, ZooKeeperClientFactory zkClientFactory)
             throws PulsarServerException {
         try {
             localZkCache = new ZookeeperCacheLoader(zkClientFactory, config.getZookeeperServers());
-            globalZkCache = new ZookeeperCacheLoader(zkClientFactory, config.getGlobalZookeeperServers());
+            globalZkCache = new GlobalZooKeeperCache(zkClientFactory, 30000, config.getGlobalZookeeperServers(),
```

Comment:
yes, then should we make that configurable also as one may want to configure the timeout in case of zk is facing performance issue.